chain/client/src/stateless_validation/state_witness_producer.rs

```diff
@@ -53,10 +53,25 @@ impl Client {
             chunk,
             transactions_storage_proof,
         )?;
-        let signed_witness = SignedEncodedChunkStateWitness::new(&witness, my_signer.as_ref());
-        metrics::CHUNK_STATE_WITNESS_TOTAL_SIZE
-            .with_label_values(&[&chunk_header.shard_id().to_string()])
-            .observe(signed_witness.witness_bytes.size_bytes() as f64);
+        let shard_id_label = chunk_header.shard_id().to_string();
+        let signed_witness = {
```

Comment:
can this block be extracted as a separate function like build_encoded_witness for readability? maybe including my_signer part above as well.

Reply:
good idea, thanks! done in 24bc64b5849fd526b0174426a5d61b4e339d8d85